omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java

```diff
@@ -353,13 +354,15 @@ public void renderAutomationPage(ModelMap model) {
 	}
 
 	private void initialiseAutomation(ModelMap model, boolean automationEnabled, List<String> postResponse) {
+		List<String> supportedPeriodTypes = Arrays.asList("Daily", "Weekly", "Monthly", "yearly", "Financial April",
+				"Financial July", "Financial Oct");
 		List<DHISOrganisationUnit> orgUnits = Context.getService(DHISConnectorService.class).getDHISOrgUnits();
 
-		model.addAttribute("mappings", Context.getService(DHISConnectorService.class).getMappings());
+		model.addAttribute("mappings", Context.getService(DHISConnectorService.class).getMappings()
+				.stream().filter(mapping -> supportedPeriodTypes.contains(mapping.getPeriodType())).collect(Collectors.toList()));
```

Comment:
```suggestion
		// add a comment
		List<type> supportedMappings = Context.getService(DHISConnectorService.class).getMappings()
						.stream()
						.filter(mapping -> supportedPeriodTypes.contains(mapping.getPeriodType()))
						.collect(Collectors.toList()));	
		model.addAttribute("mappings", supportedMappings); 
```

Note the formatting.